pkg/mapper/mapper.go

```diff
@@ -23,3 +40,30 @@ type Mapper interface {
 	Map(canonicalARN string) (*config.IdentityMapping, error)
 	IsAccountAllowed(accountID string) bool
 }
+
+func ValidateBackendMode(modes []string) []error {
+	var errs []error
+
+	validModes := sets.NewString(ValidBackendModeChoices...)
+	for _, mode := range modes {
+		if !validModes.Has(mode) {
+			errs = append(errs, fmt.Errorf("backend-mode %q is not a valid mode", mode))
+		}
+	}
+
+	for _, mode := range modes {
+		if replacementMode, ok := DeprecatedBackendModeChoices[mode]; ok {
+			logrus.Warningf("warning: backend-mode %q is deprecated, use %q instead", mode, replacementMode)
+		}
+	}
+
+	if len(modes) != len(sets.NewString(modes...).List()) {
```

Comment:
I think you can just do: 
```
if len(modes) != sets.NewString(modes...).Len() {
```

https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go#L199